File: backend/apps/academics/models/homework.py
from django.db import models


class Homework(models.Model):
    """Maps to `homework` in db_current."""

    id = models.AutoField(primary_key=True)
    class_id = models.IntegerField(blank=False, null=False, db_index=True)
    section_id = models.IntegerField(blank=False, null=False, db_index=True)
    session_id = models.IntegerField(blank=False, null=False, db_index=True)
    staff_id = models.IntegerField(blank=False, null=False, db_index=True)
    subject_group_subject_id = models.IntegerField(blank=True, null=True, db_index=True)
    subject_id = models.IntegerField(blank=True, null=True, db_index=True)
    homework_date = models.DateField()
    submit_date = models.DateField()
    marks = models.FloatField(blank=True, null=True)
    description = models.TextField(blank=True, null=True)
    assigned_to_house = models.IntegerField(blank=True, null=True)
    assigned_to_president = models.IntegerField(blank=True, null=True)
    create_date = models.DateField()
    evaluation_date = models.DateField(blank=True, null=True)
    document = models.CharField(max_length=200, blank=True, null=True)
    created_by = models.IntegerField(blank=False, null=False, db_index=True)
    evaluated_by = models.IntegerField(blank=True, null=True, db_index=True)
    created_at = models.DateTimeField(blank=True, null=True)

    class Meta:
        managed = False
        db_table = "homework"

    def __str__(self):
        return f"Homework {self.pk}"

File: frontend/src/types/academics/homework.ts
export interface Homework {
  id: number;
  class_id: number;
  section_id: number;
  session_id: number;
  staff_id: number;
  subject_group_subject_id: number | null;
  subject_id: number | null;
  homework_date: string;
  submit_date: string;
  marks: number | null;
  description: string | null;
  assigned_to_house: number | null;
  assigned_to_president: number | null;
  create_date: string;
  evaluation_date: string | null;
  document: string | null;
  created_by: number;
  evaluated_by: number | null;
  created_at: string | null;
}

export interface CreateHomeworkPayload {
  class_id: number;
  section_id: number;
  session_id: number;
  staff_id: number;
  subject_id?: number | null;
  subject_group_subject_id?: number | null;
  homework_date: string;
  submit_date: string;
  marks?: number | null;
  description?: string | null;
  create_date: string;
  created_by: number;
  document?: string | null;
  evaluation_date?: string | null;
  evaluated_by?: number | null;
}

export type UpdateHomeworkPayload = Partial<CreateHomeworkPayload>;

export interface HomeworkListFilters {
  class_id?: number;
  section_id?: number;
  session_id?: number;
  subject_id?: number;
  staff_id?: number;
  search?: string;
  page?: number;
}

export interface DailyAssignment {
  id: number;
  student_session_id: number;
  subject_group_subject_id: number;
  title: string | null;
  description: string | null;
  attachment: string | null;
  evaluated_by: number | null;
  date: string | null;
  evaluation_date: string | null;
  remark: string;
  created_at: string;
}

export interface CreateDailyAssignmentPayload {
  student_session_id: number;
  subject_group_subject_id: number;
  title?: string | null;
  description?: string | null;
  attachment?: string | null;
  evaluated_by?: number | null;
  date?: string | null;
  evaluation_date?: string | null;
  remark: string;
}

export type UpdateDailyAssignmentPayload = Partial<CreateDailyAssignmentPayload>;

export interface DailyAssignmentListFilters {
  student_session_id?: number;
  subject_group_subject_id?: number;
  evaluated_by?: number;
  search?: string;
  page?: number;
}
